tests/FSharp.Compiler.UnitTests/ProductVersion.fs

```diff
@@ -68,24 +69,6 @@ module ProductVersionTest =
         for (v, expected) in validValues() do 
             v |> productVersionToILVersionInfo |> Assert.areEqual expected
 
-    let invalidValues () =
-        [ "1.2.3.4", ILVersionInfo(1us,2us,3us,4us)
-          "1.2.3.4a", ILVersionInfo(1us,2us,3us,0us)
-          "1.2.c3.4", ILVersionInfo(1us,2us,0us,0us)
-          "1.2-d.3.4", ILVersionInfo(1us,0us,0us,0us)
-          "1dd.2.3.4", ILVersionInfo(0us,0us,0us,0us)
-          "1dd.2da.d3hj.dd4ds", ILVersionInfo(0us,0us,0us,0us)
-          "1.5.6.7.dasd", ILVersionInfo(1us,5us,6us,7us)
-          "9.3", ILVersionInfo(9us,3us,0us,0us)
-          "", ILVersionInfo(0us,0us,0us,0us)
-          "70000.80000.90000.100000", ILVersionInfo(0us,0us,0us,0us)
-          (sprintf "%d.70000.80000.90000" System.UInt16.MaxValue), ILVersionInfo(System.UInt16.MaxValue,0us,0us,0us) ]
```

Comment:
What's the rationale for deleting these other cases?

Reply:
Sorry, I might have been slightly too agressive here. In fact some of these cases are still valid, however these are incorrect:

`"1.2.c3.4", ILVersionInfo(1us,2us,0us,0us)
          "1.2-d.3.4", ILVersionInfo(1us,0us,0us,0us)
          "1dd.2.3.4", ILVersionInfo(0us,0us,0us,0us)`

In general it might be useful to think whether cases with incorrect product version should be subject to any automatic recovery logic or should rather yield an error.

Reply:
I think for the sake of keeping this PR simple it's best just to re-add the valid test cases. Generally I'd prefer yielding an error but there might have been a good reason not to.